main/autogen/autogen.cc

```diff
@@ -663,15 +664,116 @@ string ParsedFile::toMsgpack(core::Context ctx, int version) {
     return write.pack(ctx, *this);
 }
 
-void ParsedFile::classlist(core::Context ctx, vector<string> &out) {
-    auto nameToString = [&](const auto &nm) -> string { return nm.data(ctx)->show(ctx); };
+vector<string> ParsedFile::listAllClasses(const core::Context ctx) {
+    vector<string> out;
+
     for (auto &def : defs) {
         if (def.type != Definition::Class) {
             continue;
         }
         auto names = showFullName(ctx, def.id);
-        out.emplace_back(fmt::format("{}", fmt::map_join(names, "::", nameToString)));
+        out.emplace_back(fmt::format("{}", fmt::map_join(names, "::", [&ctx](const core::NameRef &nm) -> string {
+                                         return nm.data(ctx)->show(ctx);
```

Comment:
`shortName` would be much faster here

Reply:
Done in https://github.com/sorbet/sorbet/pull/1040/commits/4c4115f44b580fa87fcb7f88cbbf2aec64356b2f